### PR TITLE
Support different image-tags per component-reconciler

### PR DESCRIPTION
**Pull request by ruanxin**
- refactor the way how component reconciler get image tag

**Description**

Changes proposed in this pull request:

- integrate rma deployment into component
- refactor the way how component reconciler get image tag

**Related issue(s)**
Related issue: https://github.com/kyma-incubator/reconciler/issues/1035

### CODE REVIEW

**Comment by ruanxin** [comment]
blocked by https://github.tools.sap/kyma/management-plane-config/pull/1612

File: resources/kcp/charts/component-reconcilers/templates/deployment.yaml [hunk 1]
```diff
@@ -1,56 +1,59 @@
...
           - reconciler
           - start
-          - {{ $component }}
+          - {{ $component.name }}
           - --workspace=/tmp/reconciler
           {{- if $global.Values.options.verbose }}
           - --verbose
           {{- end }}
           - --worker-count={{ default $global.Values.options.workerCount (($values).options).workerCount }}
+          - --worker-timeout={{ default $global.Values.options.workerTimeout (($values).options).workerTimeout }}
```
Comment (Tomasz-Smelcerz-SAP) [inline resources/kcp/charts/component-reconcilers/templates/deployment.yaml:40]: Overriding these for a component requires adding the following to the .Values at the very top-level of `values.yaml` file. For example, for Istio:
```
istio:
  options:
    workerCount: 1
    workerTimeout: 2
    retriesMax: 3
    retriesDelay: 4

global:
  ...
```
Is that on purpose? I'd say, a better place for that would be "closer" to the component entry, in the place where we define it's image tag - like this:

```
   components:
      - name: "base"
        version: "38b04f8c"
      - name: "busola-migrator"
        version: "38b04f8c"
      - name: "cleaner"
        version: "38b04f8c"
      - name: "connectivity-proxy"
        version: "38b04f8c"
      - name: "istio-configuration"
        version: "38b04f8c"
      - name: "istio"
        version: "38b04f8c"
        options:
          workerCount: 1
          workerTimeout: 2
          retriesMax: 3
          retriesDelay: 4
```

What do you think?

Comment (ruanxin) [reply resources/kcp/charts/component-reconcilers/templates/deployment.yaml:40]: List here some facts based on our discussion for later refactoring:
- basically the overwrite content which Tomasz mentioned has already been offered by [management-plane-config](https://github.tools.sap/kyma/management-plane-config/blob/master/resources/control-plane/config/stage/kcp.yaml), 
- The `components` entry is under `images` at the moment, my purpose here is easy for update image tag to keep in same block, if later needs to take this entry as an overall place to configure components, should be move outside
- Found a bug for rma overwrite value, a followup fix will be created.

**Review by Tomasz-Smelcerz-SAP — APPROVED** [review]

File: resources/kcp/values.yaml [hunk 2]
```diff
@@ -2,8 +2,37 @@ global:
   defaultTenant: 3e64ebae-38b5-46a0-b1ed-9ccee153a0ae
   images:
     cloudsql_proxy_image: eu.gcr.io/kyma-project/tpi/cloudsql-docker/gce-proxy:v1.30.0-alpine-979d1ebd
+    reconciler_repo: "eu.gcr.io/kyma-project/incubator/reconciler"
     mothership_reconciler: "eu.gcr.io/kyma-project/incubator/reconciler/mothership:eeb6e71d"
     component_reconciler: "eu.gcr.io/kyma-project/incubator/reconciler/component:eeb6e71d"
```
Comment (jakobmoellersap) [inline resources/kcp/values.yaml:7]: Why is this still there when we have the component image?

**Comment by jakobmoellersap** [comment]
blocked by https://github.com/kyma-project/test-infra/pull/5306

File: resources/kcp/values.yaml [hunk 3]
```diff
@@ -2,8 +2,37 @@ global:
   defaultTenant: 3e64ebae-38b5-46a0-b1ed-9ccee153a0ae
   images:
     cloudsql_proxy_image: eu.gcr.io/kyma-project/tpi/cloudsql-docker/gce-proxy:v1.30.0-alpine-979d1ebd
+    reconciler_repo: "eu.gcr.io/kyma-project/incubator/reconciler"
     mothership_reconciler: "eu.gcr.io/kyma-project/incubator/reconciler/mothership:eeb6e71d"
```
Comment (ruanxin) [inline resources/kcp/values.yaml:6]:
```suggestion

```

**Review by tobiscr — DISMISSED** [review]

**Pull request closed without merging**